### PR TITLE
Use Rails callbacks

### PR DESCRIPTION
**Pull request by santib**
Resolves https://github.com/rootstrap/yaaf/issues/12
Resolves https://github.com/rootstrap/yaaf/issues/11

### CODE REVIEW

**Comment by fedeagripa** [comment]
@santib is the TODO for this pr or for a new one?

**Comment by santib** [comment]
> @santib is the TODO for this pr or for a new one?

For this one, I'll do it in these following days

**Comment by brunvez** [comment]
> Maybe add some happy path testing?

nvm just saw the description

File: spec/yaaf/multiple_callbacks_spec.rb [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          before_validation_counter: 1
+        )
+      end
+    end
+
+    context 'when there is a DB exception' do
+      let(:email) { nil }
+
+      it 'increments the correct values' do
+        expect { subject rescue nil }.to change { form.result }.to(
```
Comment (juanmanuelramallo) [inline spec/yaaf/multiple_callbacks_spec.rb]: 🤯

**Pull request merged**